kmip_mgmt_objects.go

```diff
@@ -0,0 +1,102 @@
+package kp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+const (
+	KMIPObjectSubPath = "kmip_objects"
+	kmipObjectType    = "application/vnd.ibm.kms.kmip_object+json"
+)
+
+type KMIPObject struct {
+	ID                string     `json:"id,omitempty"`
+	KMIPObjectType    int        `json:"kmip_object_type,omitempty"`
+	ObjectState       int        `json:"state,omitempty"`
+	CreatedByCertID   string     `json:"created_by_kmip_client_cert_id,omitempty"`
+	CreatedBy         string     `json:"created_by,omitempty"`
+	CreatedAt         *time.Time `json:"created_at,omitempty"`
+	UpdatedByCertID   string     `json:"updated_by_kmip_client_cert_id,omitempty"`
+	UpdatedBy         string     `json:"updated_by,omitempty"`
+	UpdatedAt         *time.Time `json:"updated_at,omitempty"`
+	DestroyedByCertID string     `json:"destroyed_by_kmip_client_cert_id,omitempty"`
+	DestroyedBy       string     `json:"destroyed_by,omitempty"`
+	DestroyedAt       *time.Time `json:"destroyed_at,omitempty"`
+}
+
+type KMIPObjects struct {
+	Metadata KeysMetadata `json:"metadata"`
+	Objects  []KMIPObject `json:"resources"`
+}
+
+func (c *Client) GetKMIPObjects(ctx context.Context, adapter_id string, limit, offset int, totalCount bool) (*KMIPObjects, error) {
```

Comment:
There should also be a `state` filter that can be applied to this functionality

Reply:
state filter implemented.